browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -814,7 +814,8 @@ L.CanvasTileLayer = L.Layer.extend({
 			if (!L.Browser.mobileWebkit)
 				TileManager.update(this._map.getCenter(), tileZoom);
 
-			TileManager.pruneTiles();
+			if (tileZoomChanged)
+				TileManager.pruneTiles();
```

Comment:
This change fixes `pruneTiles` being called during the course of normal document navigation.